### ISSUE TITLE
[Module Proposal]: `avm-res-eventhub-namespace`

### ISSUE THREAD

**Issue by kewalaka** [Azure/Azure-Verified-Modules#367]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Terraform

### Module Classification?

Resource Module

### Module Name

avm-res-eventhub-namespace

### Module Details

This would be the Terraform module for the Event Hub namespace & child resources, such as Event Hub itself.

### Do you want to be the owner of this module?

No

### Module Owner's GitHub Username (handle)

mbilalamjad

### (Optional) Secondary Module Owner's GitHub Username (handle)

kewalaka

**Comment by kewalaka** [comment]
yeh.. you guessed it, there's a starter available here 😂🚀

https://github.com/kewalaka/terraform-azurerm-avm-res-eventhub-namespace

just the namespace resource & a starter on the event hub child resource at the moment.

**Comment by mbilalamjad** [comment]
@matebarabas I'm happy to own this work with @kewalaka on it, could you please do the official handshake process

**Comment by matebarabas** [comment]
Hi @mbilalamjad,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by mbilalamjad** [comment]
I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER

**Comment by matebarabas** [comment]
Hi @mbilalamjad,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by mbilalamjad** [comment]
Thanks @matebarabas, will get started on this and work with @kewalaka once I return from holidays on Jan 2nd

**Comment by mbilalamjad** [comment]
Hey @kewalaka, I've created the official repo, it's under review currently and will be public soon, will keep you posted, in the meantime I'll do a review of the module in your repo next week.

https://github.com/Azure/terraform-azurerm-avm-res-eventhub-namespace

**Comment by mbilalamjad** [comment]
Hey @kewalaka, hope you're doing great, getting back with an update, the repo setup is complete and public now, I've created a dev branch and added you as contributor to it
https://github.com/Azure/terraform-azurerm-avm-res-eventhub-namespace

**Comment by kewalaka** [comment]
hi @mbilalamjad the invite says this when i try click it "The invite is invalid":

![image](https://github.com/Azure/Azure-Verified-Modules/assets/3146590/4117a197-94d1-4a78-afba-b16fc31ea238)

I'd tried it both from the email and the github portal.

**Comment by mbilalamjad** [comment]
> Hey @mbilalamjad Bilal Amjad FTE
> 
> The repo permissions aren't correct, the Azure groups need to be granted admin rights as per the spec. Can you sort please?

Thanks Matt, have fixed this

**Comment by mbilalamjad** [comment]
> Thanks Matt, have fixed this

Thanks @kewalaka will look into this

**Comment by mbilalamjad** [comment]
> hi @mbilalamjad Bilal Amjad FTE the invite says this when i try click it "This invitation is invalid":
> 
> ![image](https://private-user-images.githubusercontent.com/3146590/300128777-4117a197-94d1-4a78-afba-b16fc31ea238.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYzMjM1NjEsIm5iZiI6MTcwNjMyMzI2MSwicGF0aCI6Ii8zMTQ2NTkwLzMwMDEyODc3Ny00MTE3YTE5Ny05NGQxLTRhNzgtYWZiYS1iMTZmYzMxZWEyMzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjdUMDI0MTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTU3NzE3OTU4ZThlZTA3MGQyYzUxZTQ0MTk2MWJkNWEwOWE0NmM1MjAzNmJmN2Y4ZGMwOWFmZGI4ZmY0MTAyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VmoayogcMOtfTFRV6XrUnFNpPMgPqX6cCz93ImwSinE)
> 
> I'd tried it both from the email and the github portal, but in the meantime the following will land the code from my repo to Azure:dev
> 
> [Azure/terraform-azurerm-avm-res-eventhub-namespace#1](https://github.com/Azure/terraform-azurerm-avm-res-eventhub-namespace/pull/1)

Thanks @kewalaka will look into this.

**Comment by kewalaka** [comment]
@mbilalamjad I think dev is ready to be merged into main, but I'm not a collaborator so I can't raise the PR.

I seem to be having some issues running grept, not sure why, but I think all changes that it requires are in place.

can you raise the PR from azure:dev to azure:main please?

**Comment by mbilalamjad** [comment]
> @mbilalamjad Bilal Amjad FTE I think dev is ready to be merged into main, but I'm not a collaborator so I can't raise the PR.
> 
> I seem to be having some issues running grept, not sure why, but I think all changes that it requires are in place.
> 
> can you raise the PR from azure:dev to azure:main please?

Hey Stu, will get in touch on this later this week, thank you for your patience

**Comment by mbilalamjad** [comment]
@kewalaka getting back to you on this, I raised the PR from dev to main and merged it, also ran grept and merged a PR from it, have a look and let me how things look

**Comment by kewalaka** [comment]
there's some interest on the github issues for the repo, so i'm currently looking at realignment work with recent updates to the AVM spec, sharing just in case anyone else is thinking of doing stuff.

note there hasn't been a release yet so I'd love to get one out at the end of this exercise

**Comment by mbilalamjad** [comment]
> there's some interest on the github issues for the repo, so i'm currently looking at realignment work with recent updates to the AVM spec, sharing just in case anyone else is thinking of doing stuff.
> 
> note there hasn't been a release yet so I'd love to get one out at the end of this exercise

Hey @kewalaka, I've made some updates and put an initial v0.1.0 release out, please do feel free to raise issues/prs against the module repo for further improvements
https://registry.terraform.io/modules/Azure/avm-res-eventhub-namespace
